src/layer/vector/Canvas.js

```diff
@@ -8,9 +8,6 @@ L.Canvas = L.Renderer.extend({
 		L.Renderer.prototype.onAdd.call(this);
 
 		this._layers = this._layers || {};
-
-		// redraw vectors since canvas is cleared upon removal
-		this._draw();
```

Comment:
Humm, now that you ask me, I see there is a scenario where it's still needed: when removing and readding the rendered itself from the map.
I've removed this line because when adding a Path, it calls itself `renderer._updatePath`. And, when only dealing with adding or removing layers (not the renderer itself) from the map,  `renderer._layers` is empty at `Rendered.onAdd` time, and thus the call to `_draw` is flailing at the air.
Anyway, I'm reverting this change! :)